AK Apolo/app/src/main/java/ayds/apolo/songinfo/moredetails/model/repository/ArtistCardRepository.kt

```diff
@@ -0,0 +1,51 @@
+package ayds.apolo.songinfo.moredetails.model.repository
+
+import ayds.apolo.songinfo.moredetails.model.entities.*
+import ayds.apolo1.lastfm.LastFMArtistInfoService
+import ayds.apolo.songinfo.moredetails.model.repository.local.lastFM.LastFMLocalStorage
+import ayds.apolo.songinfo.moredetails.model.repository.local.lastFM.ArtistInfoToCardMapper
+
+interface ArtistCardRepository {
+    fun getArtistCardByName(artistName: String): Card
+}
+
+internal class ArtistCardRepositoryImpl(
+    private val artistCardLocalStorage: LastFMLocalStorage,
+    private val artistInfoTrackService: LastFMArtistInfoService,
+    private val artistInfoToCardMapper: ArtistInfoToCardMapper
+) : ArtistCardRepository {
+
+    override fun getArtistCardByName(artistName: String): Card {
+        var artistCard: Card? = artistCardLocalStorage.getArtistCardByName(artistName)
+
+        when {
+            artistCard != null -> markCardAsLocal(artistCard)
+            else -> {
+                try {
+                    val artistInfo = artistInfoTrackService.getArtistInfo(artistName)
```

Comment:
Esto seria lo mismo no?

```suggestion
artistInfoTrackService.getArtistInfo(artistName).let {
    artistCard = artistInfoToCardMapper.map(it)
    when {
        isSavedSong(artistName) -> artistCardLocalStorage.updateArtistCard(it, artistName)
        else -> artistCardLocalStorage.saveArtistCard(it, artistName)
    }
}
```